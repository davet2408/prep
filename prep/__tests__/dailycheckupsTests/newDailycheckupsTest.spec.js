import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
import db from "../../src/components/firebaseInit";
import firebase from "firebase/app";
// @ts-ignore
import Component from "../../src/components/tests/dailycheckups/NewDailyCheckups.vue";

describe("Component", () => {
  const wrapper = shallowMount(Component, {
    stubs: {
      RouterLink: RouterLinkStub
    },
    mocks: {
      $route: {
        name: "new-dailycheckups",
        params: { test_id: "id" }
      }
    }
  });

  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  }),
    //snapshot test
    test("renders correctly", () => {
      expect(wrapper.element).toMatchSnapshot();
    }),
    test("has the right amount of inputs", () => {
      expect(wrapper.findAll("input")).toHaveLength(2);
    }),
    test("all inputs have required attribute", () => {
      var inputArray = wrapper.findAll("input");
      // get if there is an input without the required property
      inputArray = inputArray.filter(input => !input.attributes("required"));
      expect(inputArray).toHaveLength(0);
    }),
    test("the title is right", ()=> {
        expect(wrapper.find('h3').text()).toBe("Add Daily Check-ups")
    }),
    test("has the correct buttons rendered", () => {
        const CancelAdd = wrapper.find("#CancelAdd");
        expect(CancelAdd).toBeDefined(); 
    }),
    test("form submits successfully", () => {
        wrapper.setMethods({ saveDailyCheckups: jest.fn() });
        wrapper.find("#submit").trigger("submit");
    
        expect(wrapper.vm.saveDailyCheckups).toHaveBeenCalled();
    }),
    test("add correct instructions", () => {
        // initialize the method
        wrapper.setMethods({ addInstruction: jest.fn()})

        wrapper.vm. addInstruction()
        const data1 = wrapper.vm.data
        const data2 = { value: "Don't eat meat" };

        expect(data2.value).toBe("Don't eat meat")
        expect(wrapper.vm.data).toBe(data1)

      });

  db.app.delete();
});
