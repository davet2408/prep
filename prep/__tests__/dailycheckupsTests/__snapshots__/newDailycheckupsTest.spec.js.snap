// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  id="new-dailycheckups"
  style="background-color: white; padding: 10px 50px 10px 50px; margin-top: 10px;"
>
  <h3>
    Add Daily Check-ups
  </h3>
   
  <div
    class="row"
  >
    <form
      class="col s12"
    >
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <input
            required="required"
            type="text"
          />
           
          <label>
            Description
          </label>
        </div>
      </div>
       
      <div
        class="row"
      >
        <div
          class="input-field col s12"
        >
          <input
            min="0"
            required="required"
            type="number"
          />
           
          <label>
            Number of Days Before Test:
          </label>
        </div>
      </div>
       
      <button
        class="btn green"
        id="add"
      >
        new instruction
      </button>
       
      <div
        class="row"
      />
       
      <button
        class="btn"
        id="submit"
        type="submit"
      >
        Submit
      </button>
       
      <a>
        <button
          class="btn grey"
          id="CancelAdd"
        >
          
      Cancel
      
        </button>
      </a>
    </form>
  </div>
</div>
`;
